### PR TITLE
Added metrics from custom `train_step`/`test_step` are now returned.

### PR DESCRIPTION
**Pull request by hertschuh**
This works the same way as in Keras 2, whereby the metrics are returned directly from the logs if the set of keys doesn't match the model metrics.

### CODE REVIEW

File: keras/src/trainers/trainer.py [hunk 1]
```diff
@@ -899,6 +899,37 @@ def _pythonify_logs(self, logs):
...
+        because it is not guaranteed to be executed the last. This method gets
+        metrics from the model directly instead of relying on the return from
+        last step function.
+
+        When the user has custom train / test step functions, the metrics
+        returned may be different from `Model.metrics`. In those instances,
+        this function will be no-op and return the logs passed in.
+
+        Args:
+          logs: A `dict` of metrics returned by train / test step function.
```
Comment (fchollet) [inline keras/src/trainers/trainer.py:916]: Add indent

File: keras/src/trainers/trainer.py [hunk 2]
```diff
@@ -899,6 +899,37 @@ def _pythonify_logs(self, logs):
...
+
+        Returns:
+          A `dict` containing values of the metrics listed in `self.metrics`
```
Comment (fchollet) [inline keras/src/trainers/trainer.py:919]: Same here

**Review by fchollet — COMMENTED** [review]
Thanks for the PR! LGTM

File: keras/src/trainers/trainer.py (hunk 1 above)
Comment (hertschuh) [reply keras/src/trainers/trainer.py]: Done

File: keras/src/trainers/trainer.py (hunk 2 above)
Comment (hertschuh) [reply keras/src/trainers/trainer.py]: Done

**Pull request merged**